### DISCUSSION TITLE
Test fails when rendering component in `beforeEach` and not inside `it`

### DISCUSSION THREAD

**Discussion by smarlhens** [vitest-dev/vitest#2069] [Q&A]
### Describe the bug

Hi there 👋🏻 ,

In one of my Vue 3 applications, I use vite, vitest & [testing-library](https://testing-library.com/docs/vue-testing-library/intro/) for testing and I have a problem with one of my unit test when I [render](https://testing-library.com/docs/vue-testing-library/api#rendercomponent-options) the component inside [beforeEach](https://vitest.dev/api/#beforeeach).

I could not find the cause of my problem.

I don't know if the error comes from your library, from [testing-library](https://testing-library.com/docs/vue-testing-library/intro/) or maybe from my code or my test implementation. If so, I apologize in advance.

Could you help me to solve this problem?

### Reproduction

I've created a repo with multiple test cases:

- ✅ test w/o [render](https://testing-library.com/docs/vue-testing-library/api#rendercomponent-options) in `beforeEach`
- ❌ test w/ [render](https://testing-library.com/docs/vue-testing-library/api#rendercomponent-options) in `beforeEach`

Repro repository : [smarlhens/vue-vite-vitest-testing-library-before-each-render-error](https://github.com/smarlhens/vue-vite-vitest-testing-library-before-each-render-error)

### System Info

```shell
System:
    OS: Linux 5.15 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish)
    CPU: (16) x64 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
    Memory: 13.66 GB / 31.08 GB
    Container: Yes
    Shell: 5.8.1 - /usr/bin/zsh
Binaries:
    Node: 16.17.0 - /usr/bin/node
    npm: 8.15.0 - /usr/bin/npm
Browsers:
    Chrome: 105.0.5195.125
    Firefox: 105.0
npmPackages:
    @vitejs/plugin-vue: 3.1.0 => 3.1.0 
    vite: 3.1.3 => 3.1.3 
    vitest: 0.23.4 => 0.23.4
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
I'd recommend asking in [Discord](https://chat.vitest.dev/)
